### ISSUE TITLE
handle when usernames change

### ISSUE THREAD

**Issue by snarfed** [snarfed/bridgy#766]
...and the rest of the profile fields we use - display name, homepage url(s), etc.

~~twitter and~~ instagram are the most important ones, to start, since we link to those silo profiles based on username, so when people change their username, our links (and often profile pictures) break.

https://brid.gy/twitter/Aluenvey is an example. they changed their twitter handle to ToDiaspora, but we didn't notice.

**Comment by mrcgrtz** [comment]
I have the same issue with Instagram where I switched my handle from `dreamseer` to `mrcgrtz`:

https://brid.gy/instagram/marcgoertz.de (which is used in the browser extension) now redirects to https://brid.gy/instagram/dreamseer instead of https://brid.gy/instagram/mrcgrtz

Is there any way I can update this on my own?

**Comment by snarfed** [comment]
@mrcgrtz sorry for the trouble! Try uninstalling and then reinstalling the plugin, and then IndieAuth your site again on Bridgy. That should clear the stored username in your browser.

**Comment by mrcgrtz** [comment]
@snarfed No troubles!

I did reinstall the extension on both Firefox and Chrome without success. The old data apparently persists in the browser and does not get deleted when uninstalling the extension.

Also, the brid.gy URL itself still redirects to my old handle:

```
$ curl https://brid.gy/instagram/marcgoertz.de
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>Redirecting...</title>
<h1>Redirecting...</h1>
<p>You should be redirected automatically to target URL: <a href="https://brid.gy/instagram/dreamseer">https://brid.gy/instagram/dreamseer</a>. If not click the link.
```

… despite having a `rel="me"` on my site with the new Instagram profile URL and `u-syndication` links on some posts.

**Comment by snarfed** [comment]
Hmm, that's surprising! Browsers are supposed to clear extensions' storage when they're uninstalled. From https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/local :

> When the extension is uninstalled, its associated local storage is cleared.

So you're logged into Instagram as mrcgrtz, not dreamseer, and after uninstalling and reinstalling the Bridgy extension, its preferences page says it thinks your Instagram username is still dreamseer?

(And apologies for the https://brid.gy/instagram/marcgoertz.de redirect. It has both usernames associated with your domain, and doesn't have a way to know which one is "correct," so it just picks the first one alphabetically.)

**Comment by mrcgrtz** [comment]
No, no, the extension itself is all fine, listing `mrcgrtz` properly.

The only thing is the brid.gy URL (or more correct: the redirect) which is why I stumbled upon this ticket in the first place. I assumed running the IndieAuth again would fix that redirect somehow, but it didn’t because—as you pointed out—it doesn’t know which username is correct.

So basically the question is: Could this lead to any troubles when using the extension or do comments and likes at `instagram.com/mrcgrtz` get properly associated to `brid.gy/instagram/mrcgrtz` instead of `brid.gy/instagram/dreamseer`?

**Comment by snarfed** [comment]
Ah, ok! Thanks for the explanation. The redirect doesn't affect backfeed, comments and likes should work correctly for both usernames as is. I'll go ahead and disable the `dreamseer` account to simplify things though. Hope that helps!

**Comment by snarfed** [comment]
Not sure why I closed this. Maybe the browser extension is a bit smarter about this for FB/IG, but I think the username change use case in Twitter, Mastodon, etc is still unsolved here.

**Comment by snarfed** [comment]
Closing. Twitter and Instagram are dead here, and Mastodon username changes are effectively new accounts.
